### PR TITLE
Command and keys

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- command bar
- keybindings using mousetrap
- search using fuse.js

### CODE REVIEW

File: components/hooks/use-bind-key/use-bind-key.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useEffect, useRef } from 'react';
+import Mousetrap from 'mousetrap';
+
+/** binds a global key to a handler */
+export function useKeyBind(
```
Comment (GiladShoham) [inline components/hooks/use-bind-key/use-bind-key.tsx:5]: why the file name is use bind key and the function is use key bind. decide if the key or the bind is first

File: extensions/command-bar/command-bar.ui.runtime.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+};
+
+/** Quick launch actions. Use the `addSearcher` slot to extend the available actions */
+export class CommandBarUI {
+  private mousetrap = new Mousetrap();
+  private commandRegistry = new Map<CommandId, CommandEntry>();
+  private commandSearcher = new CommandSearcher([]);
+
+  /** Opens the command bar */
+  open = () => {
```
Comment (GiladShoham) [inline extensions/command-bar/command-bar.ui.runtime.tsx:30]: why do you use arrow functions?

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: components/hooks/use-bind-key/use-bind-key.tsx (hunk 1 above)
Comment (KutnerUri) [reply components/hooks/use-bind-key/use-bind-key.tsx]: done

File: extensions/command-bar/command-bar.ui.runtime.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    this.searcherSlot.register(commandSearcher);
+    return this;
+  }
+
+  /**
+   * registers a command
+   */
+  addCommand(...commands: CommandEntry[]) {
+    commands.forEach(({ id: commandId }) => {
+      if (this.getCommand(commandId) !== undefined) throw new Error(`command already exists: "${commandId}"`);
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply extensions/command-bar/command-bar.ui.runtime.tsx]: replaced with an bitError, having `isUserError = true`

File: extensions/command-bar/command-bar.ui.runtime.tsx (hunk 2 above)
Comment (KutnerUri) [reply extensions/command-bar/command-bar.ui.runtime.tsx:31]: using arrow functions to bind these method to the class

File: extensions/command-bar/command-bar.ui.runtime.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    const commandEntry = this.getCommand(commandId);
+    if (!commandEntry) return undefined;
+
+    return commandEntry.handler();
+  }
+
+  /**
+   * executes a keyboard shortcut manually
+   */
+  trigger = (key: string) => {
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply extensions/command-bar/command-bar.ui.runtime.tsx:79]: same

**Pull request merged**